ADRs/ADR_TEMPLATE.md

```diff
@@ -6,6 +6,10 @@
 
 Technical Story: [description | ticket/issue URL] <!-- optional -->
 
+## Summary <!-- optional -->
+
+<!-- write this last -->
```

Comment:
@dylanlott I thought we talked about moving the "Y-statement" which is currently in the "problem and context statement" section here?